activerecord/lib/active_record/reflection.rb

```diff
@@ -624,6 +627,7 @@ def automatic_inverse_of
         # with the current reflection's klass name.
         def valid_inverse_reflection?(reflection)
           reflection &&
+            reflection != self &&
```

Comment:
☝️ I could alternatively automatically look up the plualized reflection, but I don't know if that is needed. The inverse may be set somewhere else.

Reply:
`self` seems more robust